### DISCUSSION TITLE
IPWhitelist OR HTTP Auth?

### DISCUSSION THREAD

**Discussion by b3nw** [tobychui/zoraxy#356] [Q&A] [answered]
I'm migrating over to Zoraxy from NPM and trying to replicate a setup I had where sites had an IP whitelist OR a HTTP Auth, so that if a system/user accesses outside of my defined IP ranges, they can fall back to HTTP Auth.

**Comment by tobychui** [comment, marked as answer]
Currently there is no way for you to that in Zoraxy. Though, I think with these kind of special setup you are using, keep the nginx as primary reverse proxy (generic rules w/ wildcard and proxies that rarely need to change) and use Zoraxy as a secondary proxy (i.e. only forward some domains to Zoraxy for routing, put all your testing proxy rules here) will be a better architecture in my opinion.

**Comment by b3nw** [comment]
All good, this is just default behavior in NPM if you define both http user/pw and IPs in the whitelist configuration, not sure if that is considered a special setup, but will have to just watch the project for now.

**Comment by deluxestyle** [comment]
I do have the same request. Would be nice to have such a feature in zoraxy. this is the only thing that keeps me away for the moment
